### ISSUE TITLE
How would you do this in Gatsby v1?

### ISSUE THREAD

**Issue by igghera** [pieh/gatsby-mocked-nodes#1]
Hi! I was trying to use your workaround in my website, then I realized I'm still using v1 :(
The project is too large now to migrate it to v2, and I'm not expert enough to find an alternative version of this workaround for v1.

The only issue is that `emitter` is not defined, therefore I can create mock nodes and have fix my issue, but then I can't remove those mock nodes (being `emitter.on` unavailable) creating a clash.

Can you think of any way to do this in Gatsby v1?

**Comment by pieh** [comment]
It's probably possible, just instead of `emitter`, you could use `store` (that's redux store) and listen to `SET_SCHEMA` action being dispatched - maybe something like this:
```js
store.subscribe(() => {
  const lastAction = store.getState().lastAction
  if (lastAction.type === `SET_SCHEMA `) {
    onSchemaUpdate()
  }
})
```

**Comment by pieh** [comment]
@igghera please let me know if above comment is enough to unblock You

**Comment by igghera** [comment]
Hey @pieh, first of all thanks for the prompt response, really appreciated.

I've been playing with your idea but I couldn't get it to work: the callBack function in my `subscribe` block never gets called. 

Since I'm too ignorant at this point to find the cause, I used your intuition (to leverage the store) and worked out a poor man's solution.
I'm sure there are better ways to do it, but I'm out of time and this one works for me, so here it is:

```
const allNodes = Object.keys(store.getState().nodes)
  const videoNodeExists = !!allNodes.find(
    nodeName => nodeName.indexOf('MockedNodeType') >= 0
  )

  if (!videoNodeExists) {
    boundActionCreators.createNode(mockVideoNode)
  }
```

Basically, instead of creating mocked nodes and then removing them, I conditionally create them. The condition being the absence of any node in my store with the type that I need.

As long as the store contains zero nodes with that type, I'll create one and the type will be available in my schema.
Whenever a node with that type gets created for the first (in my case using a CMS), the store will contain it, and my condition will be false, so I won't create the mocked node.

Again, I'm ignorant about graphql and how gatsby uses it, and I'm curious to hear your opinion about this hack

**Comment by pieh** [comment]
Please try https://github.com/pieh/gatsby-mocked-nodes/blob/v1/gatsby-node.js (that works in v1).

It seems like I had extra space in `SET_SCHEMA` in my comment that caused it to never be called.

As for your approach - it definitely will work, but also means that you would need to have additional checks in your code to filter out any mocked data.

**Comment by igghera** [comment]
Sorry if it took a while, my twin sons are about to be born and I was busy with that matter :)
Yes, your solution works fine! 

Unfortunately in my case the type of nodes I need are some kind of union types (they are generated by a headless CMS) and I'm not able to mock them. But yeah, for regular nodes this solution works perfectly. Thanks again for your help
